### DISCUSSION TITLE
Hide title bar when maximized

### DISCUSSION THREAD

**Discussion by agriffis** [wezterm/wezterm#3884] [General]
I'm running on GNOME/Wayland and I'd like to hide the title bar when maximized.

This seems close, but there is no `window_dims.is_maximized`

```lua
local wezterm = require 'wezterm'

local function hide_title_bar(window)
  local window_dims = window:get_dimensions()
  local overrides = window:get_config_overrides() or {}

  if window_dims.is_maximized then
    if overrides.window_decorations then
      return
    end
    overrides.window_decorations = 'NONE'
  else
    if not overrides.window_decorations then
      return
    end
    overrides.window_decorations = nil
  end
  window:set_config_overrides(overrides)
end

wezterm.on('window-resized', function(window, pane)
  hide_title_bar(window)
end)

wezterm.on('window-config-reloaded', function(window)
  hide_title_bar(window)
end)
```

Is there an easy way to detect a maximized window?

**Comment by wez** [comment]
Not currently.  You may want to consider using full screen mode instead of maximizing, as it already has this behavior.
Alt-Enter is the default to toggle full screen mode.

**Reply by agriffis** [reply]
Thanks for the reply! I did look into full screen mode, but it covers the top bar, and mutes system notifications, so I'm hoping that maximizing will work in the future.

Full disclosure: I switched back to kitty for now since wezterm kept crashing due to Wayland protocol issues. Wezterm looks cool, though—I especially like the live config reload and scripting in lua—so I'll undoubtedly try again in the future.
